src/Time/Actions/AdvanceTime.hpp

```diff
@@ -84,21 +84,27 @@ struct AdvanceTime {
            const gsl::not_null<TimeDelta*> next_time_step,
            const gsl::not_null<AdaptiveSteppingDiagnostics*> diags,
            const TimeStepper& time_stepper, const bool using_error_control) {
+          const bool new_step = next_time_id->substep() == 0;
           if (time_id->slab_number() != next_time_id->slab_number()) {
             ++diags->number_of_slabs;
             // Put this here instead of unconditionally doing the next
             // check because on the first call time_id doesn't have a
             // valid slab so comparing the times will FPE.
             ++diags->number_of_steps;
-          } else if (time_id->step_time() != next_time_id->step_time()) {
+          } else if (new_step) {
             ++diags->number_of_steps;
           }
-          if (time_step->fraction() != next_time_step->fraction()) {
-            ++diags->number_of_step_fraction_changes;
-          }
 
           *time_id = *next_time_id;
-          *time_step = next_time_step->with_slab(time_id->step_time().slab());
+
+          if (new_step) {
+            if (time_step->fraction() != next_time_step->fraction()) {
+              ++diags->number_of_step_fraction_changes;
+            }
+
+            *time_step = next_time_step->with_slab(time_id->step_time().slab());
+            *next_time_step = *time_step;
```

Comment:
Wait are the next time step and the current time step are the same?

Reply:
Yes.  What did you expect?

Reply:
Naively, I wouldn't expect them to be the same? However, I must be missing something conceptually because this was true before your changes as well. This shouldn't hold up this PR.